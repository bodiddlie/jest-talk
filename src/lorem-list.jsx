import React, {Component, PropTypes} from 'react';


export class LoremList extends Component {
    static propTypes = {
        paragraphCount: PropTypes.number.isRequired
    };

    render() {
        let lorems = [];
        for (let i = 0; i < this.props.paragraphCount; i++) {
            lorems.push(<LoremItem key={i} number={i+1} />);
        }
        return (
            <div>
                <div>
                {lorems}
                </div>
            </div>
        )
    }
}

export const LoremItem = ({number}) => (
    <div>
        <h3>Item #{number}</h3>
        <p>Here is some text.</p>
    </div>
);