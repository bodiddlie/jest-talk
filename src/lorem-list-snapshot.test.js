import React from 'react';
import renderer from 'react-test-renderer';

import {LoremList} from './lorem-list';

describe('<LoremList /> - Snapshot', () => {
  test('renders the given number of items in a snapshot', () => {
    const component = renderer.create(
      <LoremList paragraphCount={5} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});