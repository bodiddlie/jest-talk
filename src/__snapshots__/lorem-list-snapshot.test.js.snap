exports[`<LoremList /> - Snapshot renders the given number of items in a snapshot 1`] = `
<div>
  <div>
    <div>
      <h3>
        Item #
        1
      </h3>
      <p>
        Here is some text.
      </p>
    </div>
    <div>
      <h3>
        Item #
        2
      </h3>
      <p>
        Here is some text.
      </p>
    </div>
    <div>
      <h3>
        Item #
        3
      </h3>
      <p>
        Here is some text.
      </p>
    </div>
    <div>
      <h3>
        Item #
        4
      </h3>
      <p>
        Here is some text.
      </p>
    </div>
    <div>
      <h3>
        Item #
        5
      </h3>
      <p>
        Here is some text.
      </p>
    </div>
  </div>
</div>
`;
